File: dir_size.py
import os

def get_dir_size(start_path='.'):
    total_size = 0
    for dirpath, dirnames, filenames in os.walk(start_path):
        for f in filenames:
            fp = os.path.join(dirpath, f)
            # skip if it is symbolic link
            if not os.path.islink(fp):
                try:
                    total_size += os.path.getsize(fp)
                except OSError:
                    pass
    return total_size

def format_size(size):
    for unit in ['B', 'KB', 'MB', 'GB', 'TB']:
        if size < 1024.0:
            return f"{size:.2f} {unit}"
        size /= 1024.0

items = os.listdir('.')
results = []
for item in items:
    path = os.path.join('.', item)
    if os.path.isdir(path):
        size = get_dir_size(path)
    else:
        size = os.path.getsize(path)
    results.append((item, size))

results.sort(key=lambda x: x[1], reverse=True)

print(f"{'Name':<30} | {'Size':<15}")
print("-" * 50)
for name, size in results:
    print(f"{name:<30} | {format_size(size):<15}")

File: dir_size.js
const fs = require('fs');
const path = require('path');

function getDirSize(dirPath) {
  let size = 0;
  const files = fs.readdirSync(dirPath);

  for (let i = 0; i < files.length; i++) {
    const filePath = path.join(dirPath, files[i]);
    try {
      const stats = fs.statSync(filePath);
      if (stats.isFile()) {
        size += stats.size;
      } else if (stats.isDirectory()) {
        size += getDirSize(filePath);
      }
    } catch (e) {}
  }
  return size;
}

function formatSize(size) {
  const units = ['B', 'KB', 'MB', 'GB', 'TB'];
  let i = 0;
  while (size >= 1024 && i < units.length - 1) {
    size /= 1024;
    i++;
  }
  return size.toFixed(2) + ' ' + units[i];
}

const items = fs.readdirSync('.');
const results = [];

for (const item of items) {
  const itemPath = path.join('.', item);
  try {
    const stats = fs.statSync(itemPath);
    if (stats.isDirectory()) {
      results.push({ name: item, size: getDirSize(itemPath) });
    } else {
      results.push({ name: item, size: stats.size });
    }
  } catch (e) {}
}

results.sort((a, b) => b.size - a.size);

console.log(`${'Name'.padEnd(30)} | ${'Size'.padEnd(15)}`);
console.log('-'.repeat(50));
for (const res of results) {
  console.log(`${res.name.padEnd(30)} | ${formatSize(res.size).padEnd(15)}`);
}
